src/components/common/BaseButton.vue

```diff
@@ -0,0 +1,22 @@
+<script setup>
+const props = defineProps({
+  type: String,
+  size: String
+})
+</script>
+
+<template lang="">
+  <button
+    :class="{
+      'w-full px-6': size === 'large',
+      'px-4': size === 'small',
+      ' text-amber-400 border border-amber-400 bg-none': type === 'secondary',
+      ' text-white  bg-amber-400 ': type === 'primary',
+      'my-3 py-2 rounded-lg font-medium ': true
+    }"
+    @click="$emit('btn-click')"
```

Comment:
You can direclty apply @click on your custom component. Read about Fall Through Attributes (https://vuejs.org/guide/components/attrs)